test/cmd/basicresources.sh

```diff
@@ -239,6 +239,16 @@ os::cmd::expect_success_and_text "oc set env dc/test-deployment-config --list" "
 os::cmd::expect_success_and_text "oc set env dc/test-deployment-config --list --resolve" "PREFIX_FOO_BAR=secret"
 os::cmd::expect_success "oc delete secret mysecret"
 os::cmd::expect_success_and_text "oc set env dc/test-deployment-config --list --resolve" "error retrieving reference for PREFIX_FOO_BAR"
+# switch to view user to ensure view-only users can't get secrets through env var resolution
+new="$(mktemp -d)/tempconfig"
+os::cmd::expect_success "oc config view --raw > $new"
+export KUBECONFIG=$new
+project=$(oc project -q)
+os::cmd::expect_success 'oc policy add-role-to-user view view-user'
+os::cmd::expect_success 'oc login -u view-user -p anything'
+os::cmd::expect_success 'oc project ${project}'
```

Comment:
make this a `try_until_success`... role granting has to propagate to the authz cache